closed/custom/Docs.gmk

```diff
@@ -116,3 +114,187 @@ $(OPENJ9_JAVADOC_MARKER) : $(OPENJDK_JAVADOC_MARKER)
+# 1. Find all the module-summary.html files in the generated javadoc.
+# 2. For each file, look for a corresponding module-info.java file in the openj9 source tree.
+# 3. If there is no corresponding module-info.java file, replace the module description in the
+#    html file with the redirection text.
+# 4. Find and replace the summary text also in the top level index.html file.
+
+OPENJ9_ONLY_JAVADOC_SCRIPT_BODY = \
+    for file in `cd $(OPENJ9_ONLY_DOCDIR); find . -type f -name module-summary.html` ; do \
+      file=`sed 's|\./||g' <<<$$file` ; \
+      fullpath=$(OPENJ9_ONLY_DOCDIR)/$$file ; \
+      dir="$${file%/*}" ; \
+      topdir="$${file%%/*}" ; \
+      excepttopdir="$${file\#*/}" ; \
+      excepttopdir=`sed 's|-summary.html|-info.java|g' <<<$$excepttopdir` ; \
+      regex="$$topdir/.*$$excepttopdir" ; \
+      if ! grep -qe "$$regex" $(OPENJ9_ONLY_JAVADOC_FILES); then \
+        $(CP) "$$fullpath" "$(OPENJ9_ONLY_JAVADOC_TEMP)" ; \
+        $(call sed_replace_block,$(MODULE_DESCRIPTION_SECOND_LINE),$(MODULE_DESCRIPTION_LAST_LINE),$(MODULE_DESCRIPTION_MARKER_TEXT)) <"$(OPENJ9_ONLY_JAVADOC_TEMP)" >"$$fullpath"  ; \
+        $(CP) "$$fullpath" "$(OPENJ9_ONLY_JAVADOC_TEMP)" ; \
+        $(call sed_replace_regex,$(MODULE_DESCRIPTION_FIRST_LINE).*$(MODULE_DESCRIPTION_MARKER_TEXT),$(MODULE_DESCRIPTION_REPLACE_TEXT)) <"$(OPENJ9_ONLY_JAVADOC_TEMP)" >"$$fullpath"  ; \
+        $(RM) "$(OPENJ9_ONLY_JAVADOC_TEMP)" ; \
+      fi ; \
+    done ; \
+    if ! grep -q '$(OPENJDK_JDK_MANAGEMENT_SUMMARY)' $(OPENJ9_ONLY_INDEX_HTML); then \
+      $(ECHO) "$$0: $(OPENJDK_JDK_MANAGEMENT_SUMMARY_ERROR_TEXT)" ; \
+      exit 1 ; \
+    else \
+      $(CP) "$(OPENJ9_ONLY_INDEX_HTML)" "$(OPENJ9_ONLY_JAVADOC_TEMP)" ; \
+      $(call sed_replace_first,$(OPENJDK_JDK_MANAGEMENT_SUMMARY),$(OPENJDK_MODULE_REFERENCE_TEXT)) <"$(OPENJ9_ONLY_JAVADOC_TEMP)" >"$(OPENJ9_ONLY_INDEX_HTML)"  ; \
+      $(RM) "$(OPENJ9_ONLY_JAVADOC_TEMP)" ; \
+    fi ; \
+    #
+
+OPENJ9_ONLY_JAVADOC_SCRIPT_BODY_ESC = $(subst ','\'',$(OPENJ9_ONLY_JAVADOC_SCRIPT_BODY))
+
+# Create a stub package-info.java file for each package with source files in
+# j9jcl_sources but no package-info.java file. This is then used by javadoc
+# instead of the openjdk version to direct the user to the Oracle online
+# documentation for that package.
+
+# The Dependency on $(OPENJDK_JAVADOC_MARKER) ensures that the docs/api/element-list
+# file which is created during the full jdk javadoc generation exists before running
+# javadoc for just the openj9 files.
+
+$(OPENJ9_ONLY_JAVADOC_MARKER): $(OPENJDK_JAVADOC_MARKER) $(OPENJ9_ONLY_FILELIST)
+	$(call LogWarn, Generating javadoc for openj9 source files)
+	for dir in $(OPENJDK_ONLY_PKGLIST) ; do \
+ 		$(MKDIR) -p $(OPENJ9_STUBS_DIR)/$${dir} ; \
+		$(ECHO) '$(OPENJ9_ONLY_PACKAGE_STUB_DESC)' >$(OPENJ9_STUBS_DIR)/$${dir}/package-info.java ; \
+		topdir="$${dir%%/*}" ; \
+		excepttopdir="$${dir#*/}" ; \
+		package=`sed 's/\//\./g' <<<$${excepttopdir}`; \
```

Comment:
This would be easier to follow if you used `|` as above:
```
package=`sed 's|/|.|g' <<<$${excepttopdir}` ; \
```
(with space before `;`)

Reply:
Also, it should use `$(SED)` instead of `sed`.

Reply:
Changed as suggested.